beacon/beacon-cors.window.js

```diff
@@ -0,0 +1,41 @@
+// META: script=/common/utils.js
+// META: script=beacon-common.js?pipe=sub
+
+"use strict";
+
+// Execute each sample test with a cross-origin URL. If allowCors is 'true'
+// the beacon handler will return CORS headers. This test ensures that the
+// sendBeacon() succeeds in either case.
+[true, false].forEach(function(allowCors) {
+    // Implement the self.buildId extension to identify the parameterized
+    // test in the report.
+    self.buildId = function(baseId) {
+        return `${baseId}-${allowCors ? "CORS-ALLOW" : "CORS-FORBID"}`;
+    };
+
+    // Implement the self.buildBaseUrl and self.buildTargetUrl extensions
+    // to change the target URL to use a cross-origin domain name.
+    self.buildBaseUrl = function(baseUrl) {
+        return "http://{{domains[www]}}:{{ports[http][0]}}";
+    };
+    // Implement the self.buildTargetUrl extension to append a directive
+    // to the handler, that it should return CORS headers, if 'allowCors'
+    // is true.
+    self.buildTargetUrl = function(targetUrl) {
+        // Note that 'allowCors=true' is not necessary for the sendBeacon() to reach
+        // the server. Beacons use the HTTP POST method, which is a CORS-safelisted
+        // method, and thus they do not trigger preflight. If the server does not
```

Comment:
> Beacons use the HTTP POST method, which is a CORS-safelisted method, and thus they do not trigger preflight.

This is not true, the behavior here is also conditional on the type of data being transferred. Specifically, "no-cors" applies for CORS-safelisted content-types (`application/x-www-form-urlencoded`, `multipart/form-data`, or `text/plain`), other types change cors mode to "cors" and require a preflight.

Processing section covers this: https://w3c.github.io/beacon/#sec-processing-model

Could we extend these tests to account for the content-type differences?